bot/cogs/embeds.py

```diff
@@ -0,0 +1,323 @@
+import textwrap
+from collections import defaultdict
+
+from discord import Colour, Embed, TextChannel
+from discord.ext.commands import Cog, Context, command, group, ColourConverter
+
+from bot import constants
+from bot.bot import Bot
+from bot.cogs.moderation.modlog import ModLog
+from bot.decorators import with_role
+
+prefix = constants.Bot.prefix
+
+
+class Embeds(Cog):
+    """
+    A cog which allows user to send messages using the bot
+    """
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.embed_mode = defaultdict(bool)
+        self.embed = {}
+        self.embed_field_id = defaultdict(lambda: -1)
+
+    @property
+    def mod_log(self) -> ModLog:
+        """Get currently loaded ModLog cog instance."""
+        return self.bot.get_cog("ModLog")
+
+    # region: embed mode
+
+    @command(aliases=["embedcreate"])
+    @with_role(*constants.MODERATION_ROLES)
+    async def embedbuild(self, ctx: Context) -> None:
+        """Enter embed creation mode"""
+        if not self.embed_mode[ctx.author]:
+            await ctx.send(f"{ctx.author.mention} You are now in embed creation mode, use `{prefix}help Embed` for more info")
+            self.embed_mode[ctx.author] = True
+            self.embed[ctx.author] = Embed()
+        else:
+            await ctx.send(f":x: {ctx.author.mention} You are already in embed creation mode, use `{prefix}help Embed` for more info")
+
+    @command()
+    @with_role(*constants.MODERATION_ROLES)
+    async def embedquit(self, ctx: Context) -> None:
+        """Leave embed creation mode"""
+        if self.embed_mode[ctx.author]:
+            await ctx.send(f"{ctx.author.mention} You are no longer in embed creation mode, your embed was cleared")
+            self.embed_mode[ctx.author] = False
+            del self.embed[ctx.author]
+        else:
+            await ctx.send(f":x: {ctx.author.mention} You aren't in embed mode")
+
+    @command()
+    @with_role(*constants.MODERATION_ROLES)
+    async def embedshow(self, ctx: Context) -> None:
+        """Take a look at the embed"""
+        embed = await self.get_embed(ctx)
+
+        if embed is False:
+            return
+
+        await ctx.send(embed=embed)
+
+    @command()
+    @with_role(*constants.MODERATION_ROLES)
+    async def embedsend(self, ctx: Context, channel: TextChannel) -> None:
+        """Send the Embed to specified channel"""
+        embed = await self.get_embed(ctx)
+
+        if embed is False:
+            return
+
+        channel_perms = channel.permissions_for(ctx.author)
+        if channel_perms.send_messages:
```

Comment:
This if statement lacks else so when a user doesn't have permission to send messages, there will be no response from the bot.

Reply:
Resolved in 189919a5867df374ddc74441cca933c7a05a06f6